xtext/org.icyphy.linguafranca/src/org/icyphy/generator/GeneratorBase.xtend

```diff
@@ -918,74 +988,56 @@ abstract class GeneratorBase extends AbstractLinguaFrancaValidator {
-            }
-            // For warnings (vs. errors), the return code is 0.
-            // But we still want to mark the IDE.
-            if (stderr.toString.length > 0 && mode === Mode.INTEGRATED) {
-                reportCommandErrors(stderr.toString())
-            }
-            return returnCode
-        } catch (Exception ex) {
-            
-            println("--- Exception: " + ex)
-            // Try running with bash.
-            // The --login option forces bash to look for and load the first of
-            // ~/.bash_profile, ~/.bash_login, and ~/.bashrc that it finds.
-            var bashCommand = new ArrayList<String>()
-            bashCommand.addAll("bash", "--login", "-c")
-            bashCommand.addAll(command.join(" "))
-            // bashCommand.addAll("bash", "--login", "-c", 'ls', '-a')
-            println("--- Attempting instead to run: " + bashCommand.join(" "))
-            builder.command(bashCommand)
-            val stdout = new ByteArrayOutputStream()
-            val stderr = new ByteArrayOutputStream()
-            val returnCode = builder.runSubprocess(stdout, stderr)
-            if (stdout.size() > 0) {
-                println("--- Standard output from command:")
-                println(stdout.toString())
-                println("--- End of standard output.")
-            }
-            if (stderr.size() > 0) {
-                println("--- Standard error from command:")
-                println(stderr.toString())
-                println("--- End of standard error.")
-            }
-            
-            if (returnCode !== 0) {
-                if (mode === Mode.INTEGRATED) {
-                    reportCommandErrors(stderr.toString())
-                } else {
-                    reportError("Bash command returns error code " + returnCode)
-                }
-            }
-            return returnCode
+     * @param cmd The command to be executed
+     * @return A ProcessBuilder object if the command was found or null otherwise.
+     */
+    protected def createCommand(String cmd) {
+        return createCommand(cmd, #[])
+    }
+    
+    /**
+     * Creates a ProcessBuilder for a given command and its arguments.
+     * 
+     * This method ensures that the given command is executable,
+     * It first tries to find the command with 'which cmake'. If that
+     * fails, it tries again with bash. n case this fails again,
+     * it returns null. Otherwise, a correctly constructed ProcessBuilder
+     * object is returned. 
+     * 
+     * @param cmd The command to be executed
+     * @param args A list of arguments for the given command
+     * @return A ProcessBuilder object if the command was found or null otherwise.
+     */
+    protected def createCommand(String cmd, List<String> args) {
+        // Make sure the command is found in the PATH.
+        print('''--- Looking for command «cmd» ... ''')
+        val whichBuilder = new ProcessBuilder(#["which", cmd])
```

Comment:
Does `which` work on Windows? Perhaps @Soroosh129 can comment on this.

Reply:
In cmd, the `which` equivalent is ingeniously called `where`.
Although if [WSL](https://docs.microsoft.com/en-us/windows/wsl/about) is installed, `bash` is added to `C:\Windows\System32` and can be used natively.

Reply:
Alright, so we should check if we are on Windows and then use `where` for the first try and `bash ... which` as the fallback. I will change that

Reply:
I pushed some new changes. @Soroosh129 could you check how that works on Windows?

Reply:
I can confirm the new change works on Windows. There are still some bugs related to the IDE + Cpp/TS on Windows that I think are unrelated to this.

Reply:
Thanks! Could you document these bugs in a new issue? Or do we already have an issue on that?